### ISSUE TITLE
[4.0] PHP warning and more when we have a Joomla Update

### ISSUE THREAD

**Issue by infograf768** [joomla/joomla-cms#23063]
### Steps to reproduce the issue
**To test on 4.0-dev branch**
Modify version.php line 63 to `const EXTRA_VERSION = 'alpha4-dev';`
Go to database page and Fix database.
Then in Joomla Update Options choose Custom URL and add this link:
`https://update.joomla.org/core/nightlies/next_major_list.xml`
Save Options.

Display the Joomla Update page `index.php?option=com_joomlaupdate`

### Result
[13-Nov-2018 11:32:47 Europe/Berlin] PHP Warning:  Creating default object from empty value in /Applications/MAMP/htdocs/installmulti/joomla40/libraries/src/Updater/Update.php on line 299

Plus weird informations like "Unknown Error" for language packs.
<img width="1318" alt="screen shot 2018-11-13 at 11 33 02" src="https://user-images.githubusercontent.com/869724/48408607-0141a980-e73a-11e8-90e5-36acc5ff6851.png">

**Comment by brianteeman** [comment]
Is this still reproducible. I followed the instructions (except for changing alpha4 to alpha14) but I didnt get anything like your results

**Comment by Quy** [comment]
I can reproduce with the instructions provided using `alpha4-dev`. Install the French language pack and check again under `3rd Party Extensions Pre-Update Check `.

**Comment by ghost** [comment]
closed for Reasons stated above. Please reopen if needed.

**Comment by Quy** [comment]
Should be opened since it is reproducible.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23063">issues.joomla.org/tracker/joomla-cms/23063</a>.</sub>

**Comment by ghost** [comment]
too early on my side, apologize.

**Comment by brianteeman** [comment]
> I can reproduce with the instructions provided using alpha4-dev

But can you reproduce with the current j4?

**Comment by ReLater** [comment]
> But can you reproduce with the current j4?

Yes.

Due to warning: `PHP Warning:  Creating default object from empty value in /libraries/src/Updater/Update.php on line 299`

- Install a language pack, e.g. de-DE_joomla_lang_full_4.0.0v1-alpha1.zip from https://github.com/joomlagerman/joomla/releases/tag/4.0.0v1-alpha1

- Go to administrator/index.php?option=com_joomlaupdate
- Click on Check for Update
- Check error-log file

The question is now: 
See lines here: https://github.com/joomla/joomla-cms/blob/4.0.0-alpha9/libraries/src/Updater/Update.php#L294-L309

- Why is $this->currentUpdate empty?
- Is it OK that it's empty but we should check that via `if (!isset($this->currentUpdate))` before going on?

```
// For everything else there's...the default!
default:
	$name = strtolower($name);

	if (!isset($this->currentUpdate))
	{
		$this->currentUpdate = new \stdClass;
	}

	if (!isset($this->currentUpdate->$name))
	{
		$this->currentUpdate->$name = new \stdClass;
	}
```
- Or is something wrong with the language packages?

**Comment by infograf768** [comment]
Looks like there were some changes recently.
Without modifying the version. I still get the error
`PHP Warning:  Creating default object from empty value in /Applications/MAMP/htdocs/joomla40/libraries/src/Updater/Update.php on line 299`
Looks like the empty value comes from the language packs (which are 3.x packs) as I have 3 packs.
but no more `Unknown Error` in the `Compatible` column.
<img width="1313" alt="Screen Shot 2019-05-31 at 08 51 30" src="https://user-images.githubusercontent.com/869724/58687247-6630ba00-8381-11e9-995a-f04e7c17f74a.png">

After modifying the version to
`const EXTRA_VERSION = 'alpha9-dev';`
same result.

**Comment by ReLater** [comment]
I've made some additions to testing instructions in https://github.com/joomla/joomla-cms/issues/23063#issuecomment-495136350

**Comment by alikon** [comment]
if i try to install from url that language pkg i got
![Screenshot from 2019-06-26 19-30-45](https://user-images.githubusercontent.com/181681/60201612-f959f400-9848-11e9-9fcf-8efb0cd31d55.png)

**Comment by ReLater** [comment]
Did you use de-DE_joomla_lang_full_4.0.0v1-alpha1.zip (451 kB)?

There's also a FULL-Joomla package (18.1 MB). Maybe you took that?

![26-06-_2019_19-42-32](https://user-images.githubusercontent.com/20780646/60202637-4b037e00-984b-11e9-92dc-c2ad104867d4.jpg)

**Comment by alikon** [comment]
yes you are right sorry
but if i try to install https://github.com/joomlagerman/joomla/releases/download/4.0.0v1-alpha1/de-DE_joomla_lang_full_4.0.0v1-alpha1.zip
got  this

![Screenshot from 2019-06-26 19-51-52](https://user-images.githubusercontent.com/181681/60202924-e85eb200-984b-11e9-9d31-768debe03981.png)

**Comment by ReLater** [comment]
I use the current nightly. Wednesday, 26 June 2019 02:01:00 UTC and have no installation issues with that de-de package.

But I think there's a typo in plugin finder.php
`impode(` instead of `implode(`

Plugin "Extension - Finder" is not activated in my Joomla.

**Comment by ReLater** [comment]
I'll make a pr for zhe impode thing

**Comment by ReLater** [comment]
> I'll make a pr for zhe impode thing

See: https://github.com/joomla/joomla-cms/pull/25339

**Comment by alikon** [comment]
not sure it is the right way.....
give it a chance  #25352

**Comment by ghost** [comment]
Closed as having Pull Request #25352
